source/linux/background-managers/hyprpaper.js

```diff
@@ -0,0 +1,47 @@
+import childProcess from 'node:child_process';
+import {commandExists, execFile, exec} from '../util.js';
+
+export async function isAvailable() {
+	return commandExists('hyprpaper');
+}
+
+async function initialize() {
+	return new Promise(resolve => {
+		const hyprpaper = childProcess.spawn('script', ['--quiet', '--command', 'hyprpaper', '/dev/null']);
+		hyprpaper.stdout.on('data', data => {
+			if (data.toString().includes('[ERR]') || data.toString().includes('error')) {
+				throw new Error(data.toString());
+			} else {
+				resolve();
+			}
+		});
+	});
+}
+
+export async function get() {
+	const {stdout: query} = await execFile('hyprctl', ['hyprpaper', 'listactive']);
+	if (query.includes('no wallpapers active')) {
+		return;
+	}
+
+	//	Query output looks like:
+	//	 eDP-1 = /path/to/image
+	//	 HDMI-A-1 = /path/to/image
+	return query.split(' ').at(-1).replace(/(\r\n|\n|\r)/gm, '');
```

Comment:
Not clear what the `replace` call is supposed to do. But I would split on newlines first and then on ` = `.

Reply:
I used replace to remove newlines. If you split on newlines then on '=' in result there will be extra space like here: ' /path/to/image' instead of '/path/to/image'.

Reply:
You can you 'trim()' then